### PR TITLE
`azurerm_storage_account` - Support for `sas_policy`

### LINKED ISSUE #19188: Storage account - SAS token expiry upper limit
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We would like to implement the Azure built in policy for "Storage accounts should have shared access signature (SAS) policies configured" 
(https://portal.azure.com/#view/Microsoft_Azure_Policy/PolicyDetailBlade/definitionId/%2Fproviders%2FMicrosoft.Authorization%2FpolicyDefinitions%2Fbc1b984e-ddae-40cc-801a-050a030e4fbe)

However, the azurerm_storage_account resource does not allow for setting this value meaning storage accounts would imediately be non-compliant if built in this way. As a workaround we are using the azapi "type = "Microsoft.Storage/storageAccounts@2022-05-01" which does allow for this to be set in the json body but wee would prefer to use AzureRM.

### New or Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Potential Terraform Configuration

```hcl
sasPolicy = {
        expirationAction = "Log"
        sasExpirationPeriod = "00.04:00:00"
      }
```

### References

_No response_

### PR DESCRIPTION
**Pull request by magodo**
Fix #19188 

## Test

```shell
💤  TF_ACC=1 go test -v -timeout=20h ./internal/services/storage -run='TestAccStorageAccount_sasPolicy'
=== RUN   TestAccStorageAccount_sasPolicy
=== PAUSE TestAccStorageAccount_sasPolicy
=== CONT  TestAccStorageAccount_sasPolicy
--- PASS: TestAccStorageAccount_sasPolicy (199.35s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/storage       199.361s
```

### CODE REVIEW

File: internal/services/storage/storage_account_resource.go [hunk 1]
```diff
@@ -786,6 +786,30 @@ func resourceStorageAccount() *pluginsdk.Resource {
...
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"expiration_action": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								// There is no definition of this enum in the Track1 SDK due to: https://github.com/Azure/azure-sdk-for-go/issues/14589
```
Comment (katbyte) [inline internal/services/storage/storage_account_resource.go:800]: that issue was closed as wont't fix - has it been added to the swagger or should we be opening an issue on the rest api specs?

Comment (magodo) [reply internal/services/storage/storage_account_resource.go:800]: That is not a swagger issue, but a Track1 SDK issue. As they are not support track1 SDK, they closed that issue. This comment is to explain why there isn't an enum defined in the SDK, not meant to be a TODO. If there are more than one enum defined for that attribute in later API, then there will be enums defined in the SDK.

Comment (katbyte) [reply internal/services/storage/storage_account_resource.go:800]: so the swagger is correct?

also is there a reason for this to be required with a single value? we could make it optional with a default?

Comment (magodo) [reply internal/services/storage/storage_account_resource.go:800]: Yes, swagger is correct. Good idea, will make the change then.

Comment (katbyte) [reply internal/services/storage/storage_account_resource.go:806]: perfect thanks for checking - this'll be fine and be fixed by the pandora SDK then

**Review by katbyte — APPROVED** [review]
thanks @magodo - LGTM now 🦕

**Pull request merged**